### PR TITLE
Continue best practices for libpacemaker

### PR DESCRIPTION
**Pull request by kgaillot**
This moves some functions from pcmk_graph_producer.c to a new pcmk_sched_actions.c and brings them up to current guidelines.

### CODE REVIEW

**Comment by kgaillot** [comment]
@clumens , can you review this one when you have time? Low priority

File: lib/pacemaker/pcmk_sched_actions.c [hunk 1]
```diff
@@ -200,44 +200,57 @@ action_for_ordering(pe_action_t *action)
...
 
-    if (type & pe_order_implies_then) {
-        if (then->rsc) {
+    if (pcmk_is_set(order->type, pe_order_implies_then)) {
+        if (then->rsc != NULL) {
             changed |= then->rsc->cmds->update_actions(first, then, node,
-                first_flags & pe_action_optional, pe_action_optional,
-                pe_order_implies_then, data_set);
-
+                                                       first_flags & pe_action_optional,
```
Comment (clumens) [inline lib/pacemaker/pcmk_sched_actions.c:250]: You could use pcmk_is_set here too.

File: lib/pacemaker/pcmk_sched_actions.c [hunk 2]
```diff
@@ -562,34 +572,30 @@ update_action(pe_action_t *then, pe_working_set_t *data_set)
...
-
         pe_rsc_trace(then->rsc,
-                     "%s then %s: type=0x%.6x filter=0x%.6x "
-                     "(%s %s %s on %s 0x%.6x then 0x%.6x)",
-                     first->uuid, then->uuid, other->type, first_flags,
-                     action_optional_str(first_flags),
-                     action_runnable_str(first_flags),
-                     action_type_str(first_flags), action_node_str(first),
-                     first->flags, then->flags);
+                     "%s (0x%.6x) then %s (0x%.6x): type=0x%.6x node=%s",
```
Comment (clumens) [inline lib/pacemaker/pcmk_sched_actions.c:585]: There's a way to get printf to add the "0x" for you - you can do `(%#.6x)` as the format specifier.

Comment (clumens) [reply lib/pacemaker/pcmk_sched_actions.c:585]: It looks like there's a couple dozen other places we could do this same thing, so maybe a separate PR to change them all is in order.

File: include/pcmki/pcmki_sched_utils.h [hunk 3]
```diff
@@ -32,8 +32,7 @@ GList *sort_nodes_by_weight(GList *nodes, pe_node_t *active_node,
 
 extern gboolean can_run_resources(const pe_node_t * node);
 
-extern void log_action(unsigned int log_level, const char *pre_text,
-                       pe_action_t * action, gboolean details);
+void log_action(const char *pre_text, pe_action_t *action, gboolean details);
```
Comment (clumens) [inline include/pcmki/pcmki_sched_utils.h]: Might as well change the last arg to `bool` and change the couple places it's being called with TRUE/FALSE while you're at it.  Right now, it's called with both.

Comment (clumens) [reply include/pcmki/pcmki_sched_utils.h]: Ah, already done two patches later.  I only looked one patch later to check on this.

**Comment by clumens** [comment]
Looks good to me.

File: lib/pacemaker/pcmk_sched_actions.c (hunk 2 above)
Comment (kgaillot) [reply lib/pacemaker/pcmk_sched_actions.c:585]: Do you know if that is universally supported?

Comment (clumens) [reply lib/pacemaker/pcmk_sched_actions.c:585]: It is specified in the standard at least as far back as C99 (ISO9899:1999).

**Pull request merged**